### PR TITLE
Fixes #6700: Promises.st can't load ncf on windows

### PR DESCRIPTION
**Pull request by peckpeck**
http://www.rudder-project.org/redmine/issues/6700

### CODE REVIEW

**Comment by peckpeck** [comment]
PR updated

File: techniques/system/common/1.0/promises.st [hunk 1]
```diff
@@ -115,6 +115,15 @@ bundle common va
 
       "ncf_local_inputs_50_60" slist => splitstring(execresult("${list_compatible_inputs} ${sys.cf_version} ${ncf_path}/local 50_techniques 60_services", "useshell"), "\n", 10000);
 
+    windows::
+      # Windows agent only exists in 3.6 version, so as a shortcut, we can load all files of ncf
+      # It should be replaced by a proper script, as explained in #5659
+      "ncf_common_inputs_10_40" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"Get-ChildItem -Path 'c:\Program Files\Rudder\var\ncf\common' -Recurse |  Select-Object -ExpandProperty FullName | Select-String -Pattern '(10_ncf_internals-20_cfe_basics|30_generic_methods|40_it_ops_knowledge).*cf$'  \"", "useshell"), "\n", 10000);
```
Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: why 10_ncf_internals-20_cfe_basics ?

File: techniques/system/common/1.0/promises.st [hunk 2]
```diff
@@ -115,6 +115,15 @@ bundle common va
...
+      "ncf_common_inputs_50_60" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"Get-ChildItem -Path 'c:\Program Files\Rudder\var\ncf\common' -Recurse |  Select-Object -ExpandProperty FullName | Select-String -Pattern '(50_techniques|60_service).*cf$'  \"", "useshell"), "\n", 10000);
+      "ncf_local_inputs_10_40" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"Get-ChildItem -Path 'c:\Program Files\Rudder\var\ncf\local' -Recurse |  Select-Object -ExpandProperty FullName | Select-String -Pattern '(10_ncf_internals-20_cfe_basics|30_generic_methods|40_it_ops_knowledge).*cf$'  \"", "useshell"), "\n", 10000);
```
Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: why 10_ncf_internals-20_cfe_basics ?

**Comment by peckpeck** [comment]
PR updated

File: techniques/system/common/1.0/promises.st [hunk 3]
```diff
@@ -115,6 +115,15 @@ bundle common va
...
+      "ncf_common_inputs_50_60" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"Get-ChildItem -Path 'c:\Program Files\Rudder\var\ncf\common' -Recurse |  Select-Object -ExpandProperty FullName | Select-String -Pattern '(50_techniques|60_service).*cf$'  \"", "useshell"), "\n", 10000);
+      "ncf_local_inputs_10_40" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"Get-ChildItem -Path 'c:\Program Files\Rudder\var\ncf\local' -Recurse |  Select-Object -ExpandProperty FullName | Select-String -Pattern '(10_ncf_internals|20_cfe_basics|30_generic_methods|40_it_ops_knowledge).*cf$'  \"", "useshell"), "\n", 10000);
+      "ncf_common_inputs_50_60" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"Get-ChildItem -Path 'c:\Program Files\Rudder\var\ncf\local' -Recurse |  Select-Object -ExpandProperty FullName | Select-String -Pattern '(50_techniques|60_service).*cf$'  \"", "useshell"), "\n", 10000);
```
Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: it should be ncf_local_inputs_50_60

**Comment by peckpeck** [comment]
PR updated

**Pull request merged**